### DISCUSSION TITLE
Migration advice....

### DISCUSSION THREAD

**Discussion by Vinisz** [zwave-js/node-red-contrib-zwave-js#135] [Q&A]
Ok, so, I tested this on an RPI with and UZB zwave controller and that all looked ok....
But, Now I want to migrate from Zwave.me (ZWAY) , to this....
Question is, what is the best way to do this and, even better, is there a "soft" way to do it (that would require node-red and zway to run in parralel, which seems not so be working cause both try to claim the controller)
I can of course just let the controller query and inventory, so no need to do anything on the "zwave" part (at least not on the devices) but I have over 150 zwave devices connected here, so it's not like I will be able to do it all in just couple of hours, it will take days, and functionality needs still to be there....

**Comment by marcus-j-davies** [comment]
Hey

Sadly,

There is a no way to 're-format' the devices database from one application to another.
Every application will have its own way for storing device data.

zwave-js stores its data in a different format to zwave.me - I don't think there is a way.
if the devices are mains powered, and in good range - it might not actually take that long.

As you said, you don't need to disconnect the devices from the controller itself, but the interviews still need to happen initially so, zwave-js knows about there abilities.

**Reply by Vinisz** [reply]
Yeah, I guessed that, that's why I'm aiming to run both solutions in parralel, but that also will most-likely not work...
What I can try to do is, I have another UZB controller here, I can make that part of the network as secondary controller, attached to node-red  maybe and soft- migrate that way

**Reply by marcus-j-davies** [reply]
This will still require the interview on the zwave-js side.
zwave-js (I think) will act as the primary controller.

@AlCalzone,
Whats your take on migrating 150 devices from one software solution (zwave.me) to zwave-js (node-red)
There is a spare USB controller at hand, that may help?

(and a spare PI)

**Reply by AlCalzone** [reply]
Z-Wave JS expects to be the primary controller, so "joining" it into an existing network won't be possible.
IMO it depends what causes less pain. Either exclude devices one by one and re-include them in Z-Wave JS. Or update all automations at once.

**Reply by marcus-j-davies** [reply]
I was just betting on just allow the interviews to do their thing, when the controller is attached to the 'new software'?

**Reply by Vinisz** [reply]
@marcus-j-davies , yes, if secondary controller is not possible this is the best... but... we can also turn it around of course, I can make zwave-js the primary controller and z-way the secondary... If that works, then commands (so normal "house" operations) can just go to the secondary and the "new" primary can be used to do all the interview stuff and after that, adjust my flows one-by-one in node-red (adjusting all those flows is where all the work is....)

**Reply by Vinisz** [reply]
> Z-Wave JS expects to be the primary controller, so "joining" it into an existing network won't be possible. IMO it depends what causes less pain. Either exclude devices one by one and re-include them in Z-Wave JS. Or update all automations at once.

Excluding and including is a total nightmare.... I have been to that process once and will never, ever do it again.... Works with 10 devices maybe, but not with this amount

**Comment by marcus-j-davies** [comment]
If you do embark on this journey, you will have to keep us updated.
150 devices being interviewed (albeit in an orderly fashion) will certainly be of interest to us 😆

**Reply by AlCalzone** [reply]
FYI, 150 devices would be the largest network so far, but not by far. There's one in the 120-125 range already.

**Reply by Vinisz** [reply]
Just checked, it's 119 "real" z-wave devices actually, (some others are wifi/homekit based) and there's a view virtual ones... so that makes the +/- 150 in total... but 119 zwave only is already a lot I think...

**Comment by Vinisz** [comment]
And a spare Rpi ;-)
